crates/pallet-domains/src/lib.rs

```diff
@@ -143,9 +153,16 @@ mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
-        // TODO: proper weight
         #[pallet::call_index(0)]
-        #[pallet::weight((10_000, Pays::No))]
+        #[pallet::weight(
+            if signed_opaque_bundle.domain_id().is_system() {
+                T::WeightInfo::submit_system_bundle(
+                    signed_opaque_bundle.bundle.receipts.len() as u32
+                )
+            } else {
+                T::WeightInfo::submit_core_bundle()
+            }
```

Comment:
Only the system domain receipts are handled within pallet-domains, core domain receipts are handled in pallet-domain-registry.